### DISCUSSION TITLE
Unparsable structured data issue

### DISCUSSION THREAD

**Discussion by bendracup1** [Shopify/dawn#3196] [Q&A]
Hi all,

I'm getting this error in Google Search Console: Unparsable structured data - Parsing error: Missing ',' or ']' in array declaration

It looks like this issue is caused because there are commas missing in the product schema

Image 27 is what it currently looks like and Image 28 is what it should be

Granted, the website is still on the 6.0.2, however, from looking at the release notes for updated versions, I cant see that this issue has been resolved.

Has anyone else had this issue? 

![image (27)](https://github.com/Shopify/dawn/assets/156581738/6304ff58-4f10-456b-adda-f886b87be3e5)
![image (28)](https://github.com/Shopify/dawn/assets/156581738/8dafffb8-950b-49f3-befc-ca99202a9f10)

**Comment by rsusano** [comment]
Hi! That Google Search Console error (“Missing ',' or ']' in array declaration”) almost always means your JSON-LD output contains invalid JSON, most commonly a missing comma between array elements.

In Liquid/JSON-LD, the reliable fix is:
- when you output arrays with `{% for %}`, insert commas using `unless forloop.last` (or `if forloop.first`/`last`) instead of trying to hand-type commas in many places
- avoid manual string concatenation; prefer building objects/arrays via Liquid and outputting with `| json` where possible

Example pattern for arrays:
[ {% for item in items %} {{ item | json }}{% unless forloop.last %},{% endunless %} {% endfor %} ]


Then validate the final JSON-LD in the browser (View source / copy JSON-LD) with a JSON parser or in Search Console again.

If you share the snippet/file where Dawn is generating the product schema in your theme (often a product JSON-LD snippet), I can point out exactly where the comma is missing.
